src/main/java/org/spongepowered/api/event/stats/AchievementEvent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.event.stats;
+
+import org.spongepowered.api.entity.player.Player;
+import org.spongepowered.api.event.entity.living.player.PlayerEvent;
+import org.spongepowered.api.stats.achievement.Achievement;
+import org.spongepowered.api.util.event.Cancellable;
+
+/**
+ * Represents an event that is called when a {@link Player} earns an
+ * {@link Achievement}.
```

Comment:
> The event would be for every time that the player is granted the achievement.

Granted for the first time? Granted another time? If we don't have a getter here we should mention in the javadocs that the event is only called if the achievement is granted to the player the **first** time/ if he did not have it before. Otherwise we might get inconsistencies with other servers implementing the sponge API/behaviour differently.

> No need for a current value getter imo if you do need it you can always get it from the player.

No, the `Player` does not have a getter for that too.